app/lib/activitypub/activity/emoji_react.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class ActivityPub::Activity::EmojiReact < ActivityPub::Activity
+  def perform
+    original_status = status_from_uri(object_uri)
+    name = @json['content']
+    return if original_status.nil? ||
+              !original_status.account.local? ||
+              delete_arrived_first?(@json['id']) ||
+              @account.reacted?(original_status, name)
```

Comment:
Is it sufficient for `reacted?` to be called with merely a name? After all, the name does not include the domain when it's a custom emoji, as it stands currently.

Reply:
The issue is, `reacted?` in activity.rb does check for name as it's saved in the database... and the database never saves the domain part anywhere.

Reply:
It should

Reply:
Ooooh, you're right, sorry, I misunderstood the code.
Yeah, `reacted?` needs an optional emoji attribute.

Reply:
Wait, no, name is case sensitive and since even foreign custom emoji are always treated as coming from the reactors domain, the same account can't have two different reaction with the same name attached to a toot.

btw this also looks for `:shortcode:` in the database.